### ISSUE TITLE
performance problem with large message data

### ISSUE THREAD

**Issue by Skalkaz** [zendesk/radar#141]
I made a mini test and it seems radar is getting slow with data size. The two test cases.

1) Brute-force test: I upload the data from client to a web server (no radar here). The server responses then I send a radar message to the other client "available, donwload it". The client download it and send a radar message back: "done".

2) Radar test: I send a radar message with the data. And the other client responses "done".

Both tests run on my local machine, no internet connection used. I measured the total time.

Results:
very small data size: 1) 176 ms 2) 25ms
1M data size:            1) 176 ms 2) 1000ms

Why is it so slow? Is it normal?

**Comment by lucasefe** [comment]
@Skalkaz Can you explain how you are performing those tests? I would like to understand the "I send a radar message with the data" part of the test specifically. 

Thanks

**Comment by Skalkaz** [comment]
@lucasefe With publish-subscribe method. The data is JSON.
RadarClient.message('my/event').publish({str: 'sssssssss'});

Now I also checked with sync method, similar result.

It is a bigger project in coffeescript. I don't have test code.

Fascinating reaction speed. Sorry for my late respnse, if you need more info or more test, tell me. I will try to answer faster.

Thanks.

**Comment by Skalkaz** [comment]
I tested more and I figured out: 
1) the total time is quazi-linear with data size
2) it turn out that the console.log is responsible for the 2/3 of the slowness. So this is my fault!

now the time varies:
size, time
1000 character         --> 15ms
1000000 character   --> 300ms
10000000 character --> 3900ms

So it is more than likely that the rest slowness is also my fault. I keep searching for it.

Still I wonder is it possible that at given size it is worth to use the brute-force method?
